### DISCUSSION TITLE
case insensitive match with `-match-regex` / `-mr` option

### DISCUSSION THREAD

**Discussion by gprime31** [projectdiscovery/httpx#1151] [Q&A] [answered]
is there a way to add a flag to ignore case (example: grep --ignore-case "blah" file) to the match regex flag.
or is there a way to do this already and I missed it?

**Comment by gwen001** [comment, marked as answer]
Since the value you provide is passed to the Go function `regexp.Compile`, you can use the following trick:

`cat urls.txt | httpx -mr "(?i)whatever"`
`cat urls.txt | httpx -mr "(?i)(something|blablabla|guess what)"`

Tested, it works.

https://github.com/google/re2/wiki/Syntax

**Comment by gprime31** [comment]
> Since the value you provide is passed to the Go function `regexp.Compile`, you can use the following trick:
> 
> `cat urls.txt | httpx -mr "(?i)whatever"` `cat urls.txt | httpx -mr "(?i)(something|blablabla|guess what)"`
> 
> Tested, it works.
> 
> https://github.com/google/re2/wiki/Syntax

Thanks for the tip!
